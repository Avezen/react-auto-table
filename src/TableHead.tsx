import React from 'react';

interface TableHeadProps {
    sortBy: string;
    setSortBy: any;
    sortDirection: number;
    tableHead: string[];
}

export const TableHead = ({
                              sortBy,
                              setSortBy,
                              sortDirection,
                              tableHead,
                          }: TableHeadProps) => (
    <thead>
    <tr>
        {tableHead.length > 0 && (
            <React.Fragment>
                {tableHead.map((item: string, key: number) => (
                    <th key={key} onClick={() => setSortBy(item.replace(/\s/g, ''))}>
                        {item}
                        {item.replace(/\s/g, '') === sortBy &&
                        (sortDirection === 1 ? <i className={'arrow down'}></i> : <i className={'arrow up'}></i>)}
                    </th>
                ))}
                <th>Actions</th>
            </React.Fragment>
        )}
    </tr>
    </thead>
);
