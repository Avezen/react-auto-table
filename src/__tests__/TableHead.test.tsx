import React from "react";
import {render, fireEvent} from "@testing-library/react";
import {TableHead, TableHeadProps} from "../TableHead";
import renderer from 'react-test-renderer';
import {mount} from "enzyme";

function renderTableHead(props: Partial<TableHeadProps> = {}) {
    const defaultProps: TableHeadProps = {
        sortBy: 'Id',
        setSortBy() {
            return;
        },
        sortDirection: 1,
        tableHead: ['Id', 'Name'],
    };

    return <TableHead {...defaultProps} {...props} />;
}

describe("<TableHead />", () => {
    test("should render component", async () => {
        const findByTestId = renderTableHead(),
            TableHeadComponent = renderer.create(findByTestId).toJSON();

        expect(TableHeadComponent).toMatchSnapshot();
    });

    test("should click to change direction", async () => {
        const setSortBy = jest.fn(),
            findByTestId = renderTableHead({setSortBy}),
            TableHeadComponent = mount(findByTestId).find('th').first();

        TableHeadComponent.simulate('click');
        expect(setSortBy).toHaveBeenCalledTimes(1);
    });

    test("should click to change sort by", async () => {
        const setSortBy = jest.fn(),
            findByTestId = renderTableHead({setSortBy}),
            TableHeadComponent = mount(findByTestId).find('th').at(1);

        TableHeadComponent.simulate('click');
        expect(setSortBy).toHaveBeenCalledTimes(1);
    });

    test("should not click on actions column", async () => {
        const setSortBy = jest.fn(),
            findByTestId = renderTableHead({setSortBy}),
            TableHeadComponent = mount(findByTestId).find('th').at(2);

        TableHeadComponent.simulate('click');
        expect(setSortBy).toHaveBeenCalledTimes(0);
    });
});
