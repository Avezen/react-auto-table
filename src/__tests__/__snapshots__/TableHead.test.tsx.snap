// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHead /> should render component 1`] = `
<thead>
  <tr>
    <th
      onClick={[Function]}
    >
      Id
      <i
        className="arrow down"
      />
    </th>
    <th
      onClick={[Function]}
    >
      Name
    </th>
    <th>
      Actions
    </th>
  </tr>
</thead>
`;
